tests/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java

```diff
@@ -247,4 +248,31 @@ public void cancelWrite() throws Exception {
       Assert.assertEquals(0, worker.getUsedBytes());
     }
   }
+
+  @Test
+  public void getStatusBeforeClose() throws Exception {
+    AlluxioURI path = new AlluxioURI(PathUtils.uniqPath());
+    try (FileOutStream os = mFileSystem.createFile(path, CreateFilePOptions.newBuilder()
+            .setWriteType(mWriteType.toProto()).setRecursive(true).build())) {
+      for (int i = 0; i < 3; i++) {
+        os.write(BufferUtils.getIncreasingByteArray(i * BLOCK_SIZE_BYTES, BLOCK_SIZE_BYTES));
+        // Fetch file status when the stream is still open
+        URIStatus status = mFileSystem.getStatus(path);
+        if (!mWriteType.isThrough()) {
+          // When the writeType is THROUGH, we only see the blocks when the file is committed
+          // When the file is not committed, we do not see the FileBlockInfo
```

Comment:
Before this change, we see `FileBlockInfo` but that does not necessarily match the number of block IDs. And all `FileBlockInfo` will have 0 length.
After this change, we see 0 `FileBlockInfo` before the file is committed.